tests/functional/olp-cpp-sdk-core/DirTest.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <olp/core/utils/Dir.h>
+#include <fstream>
+#if !defined(_WIN32) || defined(__MINGW32__)
+#include <unistd.h>
+#else
+#include <windows.h>
+#endif
+
+namespace {
+using Dir = olp::utils::Dir;
+
+#if defined(_WIN32) && !defined(__MINGW32__)
+const char kSeparator[] = "\\";
+#else
+const char kSeparator[] = "/";
+#endif
+
+void create_file(const std::string& path, size_t size) {
+  std::ofstream file(path.c_str(), std::ios::out | std::ios::binary);
+  file.seekp(size - 1);
+  file.write("", 1);
+  file.close();
+}
+
+void create_directory(const std::string& path) { Dir::Create(path); }
+
+void create_symlink(const std::string& to, const std::string& link) {
+#if !defined(_WIN32) || defined(__MINGW32__)
+  symlink(to.c_str(), link.c_str());
+#else
+  CreateSymbolicLink(to.c_str(), link.c_str(), 0);
+#endif
+}
+
+std::string join(std::string arg) { return arg; }
+
+template <typename... Ts>
+std::string join(std::string arg1, std::string arg2, Ts... args) {
+  return join(std::string(arg1) + kSeparator + std::string(arg2), args...);
+}
```

Comment:
According to our coding style method/function should be CamelCase, not snake_case.

Reply:
Done.